CMakeLists.txt

```diff
@@ -282,6 +282,31 @@ endif()
 # enable CTest
 enable_testing ()
 
+# Macro to check dependencies before adding tests
+
+macro(CHECK_DEPENDENCIES TEST_DEPENDENCY_PROGRAMS TEST_DEPENDENCY_LIBRARIES)
+    foreach(PROG ${TEST_DEPENDENCY_PROGRAMS})
+        find_program(${PROG}_FOUND ${PROG})
+        if (${PROG}_FOUND)
+            message(STATUS "Found program ${PROG} at ${${PROG}_FOUND}")
+        else()
+            message(WARNING "Missing program ${PROG}, disabling relevant tests."
+            " Please install ${PROG} and ensure it is in your PATH.")
+            set(TEST_DEPENDENCY_PRESENT FALSE)
+        endif()
+    endforeach()
+
+    foreach(LIB ${TEST_DEPENDENCY_LIBRARIES})
+        find_library(${LIB}_FOUND ${LIB} HINTS "${CMAKE_CURRENT_SOURCE_DIR}/runtime/usr/lib64/")
```

Comment:
The HINTS command just gives the first folder to look for. If it does not find the library it looks into other locations. I can remove the HINTS if you want. A more general solution for HINTS directories would be rather difficult to implement. I think we should leave it on find_library to search for the appropriate location.

Reply:
Let's remove it until we really needed. Normal CMake should have enough information to look in the appropriate folders. 

Since this is a function you can make it part of the parameters (with an explanation what the parameter is for).

Reply:
ok I have removed the HINTS part from the default implementation. I have modified the function to accept HINTS as a optional parameter. I have added comments with proper explanation.